blockchain/chain/src/store/chain_store.rs

```diff
@@ -88,28 +91,33 @@ impl<'a> ChainStore<'a> {
         }
         Ok(())
     }
+
     /// Returns genesis blockheader from blockstore
     pub fn genesis(&self) -> Result<BlockHeader, Error> {
-        let bz = self.db.read(self.genesis.key())?;
+        // TODO change data store for pulling genesis
+        let bz = self.db.read("gen_block")?;
```

Comment:
If we are not having the genesis cid key within the `ChainStore` struct and are storing the genesis with the key `gen_block` then the function `set_genesis` needs to be updated to store the genesis with that same key.

Reply:
The difference is the genesis block needs to be pulled from storage from a given static string, not by a cid. When starting the node, there is no way to know what the genesis Cid is to pull from db

Reply:
Which is fine but `set_genesis` will still need to be updated to reflect the static string as the key no?

Reply:
oh yes, I forgot to add a todo or mention it. We basically need some seperate data store seperate from the chain store (or part of that type) to save genesis as. I just haven't thought about how exactly this will be done

Reply:
Updated and added test, let me know if changes work and I'll resolve this